packages/template-retail-react-app/app/components/cancel-order-modal/index.jsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage} from 'react-intl'
+import {
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalFooter,
+    ModalBody,
+    ModalCloseButton,
+    Button,
+    Text
+} from '@chakra-ui/react'
+
+/**
+ * A Modal for requesting order cancellation
+ */
+const CancelOrderModal = ({
+    isOpen,
+    onClose,
+    order = null,
+    onRequestCancellation = null,
+    ...props
+}) => {
+    const handleRequestCancellation = () => {
+        if (onRequestCancellation) {
```

Comment:
Should `onRequestCancellation` be falsy at all? That means we don't do anything, which sounds wrong?